game.tl

```diff
@@ -213,14 +219,37 @@ local function clear_items(hitting_ground: {PressureBlock})
 	end
 end
 
+local function find_closest_non_hard_position(pos: Point): Point
+	-- Snap to nearest grid tile
+	local x = math.floor((pos.x - PLAYER_AREA.x) / BLOCK_SIZE + 0.5) * BLOCK_SIZE + PLAYER_AREA.x
+	local y = math.floor((pos.y - PLAYER_AREA.y) / BLOCK_SIZE + 0.5) * BLOCK_SIZE + PLAYER_AREA.y
+	return {x = x, y = y}
+end
+
+local function handle_bomb_drop_input(player: Player)
+	if input_pressed_once(player.id, JOYPAD_R) then
```

Comment:
In our previous discussion we thought it would be sufficient to simply check if a bomb was underneath the player and allow for holding JOYPAD_R here. This is fine if you stand still, but if you''re moving and you're on the edge of a tile and intend to only drop 1 bomb, you can sometimes accidentally put down 2. This is why `input_pressed_once` is necessary.